code/modules/modular_computers/hardware/rfid_programmer.dm

```diff
@@ -0,0 +1,24 @@
+/obj/item/stock_parts/computer/rfid_programmer
+	name = "RFID programmer"
+	desc = "Special interface for programming integrated devices or editing access controls. Necessary for some programs to work properly."
+	power_usage = 10 // Watts
+	critical = 0
+	icon_state = "netcard_basic"
+	hardware_size = 1
+	origin_tech = list(TECH_DATA = 4)
+	// usage_flags = PROGRAM_PDA & PROGRAM_TABLET
+
+	var/obj/linked_device			// What device we're editing the permissions for.
+
+/obj/item/stock_parts/computer/rfid_programmer/proc/link_device(var/mob/usr, var/obj/target)
+	linked_device = target
```

Comment:
this kinda stuff can result in GC failures because device holds the reference to the target.
I think here the easiest way would be to use weakrefs
so here it'd be `weakref(target)`
and in get_device() you'd do 
var/obj/result = linked_device.resolve()
then check sanity/distance and return 'result'